src/sentry/api/authentication.py

```diff
@@ -69,6 +71,38 @@ def authenticate_credentials(self, userid, password):
         return (AnonymousUser(), key)
 
 
+class ClientIdSecretAuthentication(QuietBasicAuthentication):
+    """
+    Authenticates a Sentry Application using its Client ID and Secret
+
+    This will be the method by which we identify which Sentry Application is
+    making the request, for any requests not scoped to an installation.
+
+    For example, the request to exchange a Grant Code for an Api Token.
+    """
+
+    def authenticate(self, request):
+        data = json.loads(request.body)
```

Comment:
Do you need to handle invalid JSON?

Reply:
FWIW I would prefer if we abstract around loading JSON data somehow. We do this very inconsistently at the moment.

Reply:
And also `sentry.utils.json` over normal `json`.

Reply:
Yes, to all three of these things, heh.